mobile/lib/shared/ui/immich_image.dart

```diff
@@ -174,3 +110,103 @@ class ImmichImage extends StatelessWidget {
     }
   }
 }
+
+class _ImmichImageState extends State<ImmichImage> {
+  // Creating the Uint8List from the List<bytes> during each build results in flickers during
+  // the fade transition. Calculate the hash in the initState and cache it for further builds
+  Uint8List? thumbHashBytes;
+  static const _placeholderDimension = 300.0;
+
+  @override
+  void initState() {
+    super.initState();
+    if (widget.asset?.thumbhash != null) {
+      final bytes = Uint8List.fromList(widget.asset!.thumbhash!);
```

Comment:
it should be safe to cast the thumbhash to an Uint8List (after my other comment is addressed). Isar also returns its `List<byte>` as Uint8List. This removes 1 unnecessary array copy operation.